x-pack/plugins/stack_alerts/server/alert_types/es_query/alert_type.ts

```diff
@@ -0,0 +1,303 @@
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { Logger } from 'src/core/server';
+import { ESSearchResponse } from '../../../../../typings/elasticsearch';
+import { AlertType, AlertExecutorOptions } from '../../types';
+import { ActionContext, EsQueryAlertActionContext, addMessages } from './action_context';
+import {
+  EsQueryAlertParams,
+  EsQueryAlertParamsSchema,
+  EsQueryAlertState,
+} from './alert_type_params';
+import { STACK_ALERTS_FEATURE_ID } from '../../../common';
+import { ComparatorFns, getHumanReadableComparator } from '../lib';
+import { parseDuration } from '../../../../alerts/server';
+import { buildSortedEventsQuery } from '../../../common/build_sorted_events_query';
+
+export const ES_QUERY_ID = '.es-query';
+
+const DEFAULT_MAX_HITS_PER_EXECUTION = 1000;
+
+const ActionGroupId = 'query matched';
+const ConditionMetAlertInstanceId = 'query matched';
+
+export function getAlertType(
+  logger: Logger
+): AlertType<EsQueryAlertParams, EsQueryAlertState, {}, ActionContext, typeof ActionGroupId> {
+  const alertTypeName = i18n.translate('xpack.stackAlerts.esQuery.alertTypeTitle', {
+    defaultMessage: 'ES query',
+  });
+
+  const actionGroupName = i18n.translate('xpack.stackAlerts.esQuery.actionGroupThresholdMetTitle', {
+    defaultMessage: 'Query matched',
+  });
+
+  const actionVariableContextDateLabel = i18n.translate(
+    'xpack.stackAlerts.esQuery.actionVariableContextDateLabel',
+    {
+      defaultMessage: 'The date the alert met the threshold condition.',
+    }
+  );
+
+  const actionVariableContextValueLabel = i18n.translate(
+    'xpack.stackAlerts.esQuery.actionVariableContextValueLabel',
+    {
+      defaultMessage: 'The value that met the threshold condition.',
+    }
+  );
+
+  const actionVariableContextHitsLabel = i18n.translate(
+    'xpack.stackAlerts.esQuery.actionVariableContextHitsLabel',
+    {
+      defaultMessage: 'The documents that met the threshold condition.',
+    }
+  );
+
+  const actionVariableContextMessageLabel = i18n.translate(
+    'xpack.stackAlerts.esQuery.actionVariableContextMessageLabel',
+    {
+      defaultMessage: 'A pre-constructed message for the alert.',
+    }
+  );
+
+  const actionVariableContextTitleLabel = i18n.translate(
+    'xpack.stackAlerts.esQuery.actionVariableContextTitleLabel',
+    {
+      defaultMessage: 'A pre-constructed title for the alert.',
+    }
+  );
+
+  const actionVariableContextIndexLabel = i18n.translate(
+    'xpack.stackAlerts.esQuery.actionVariableContextIndexLabel',
+    {
+      defaultMessage: 'The index the query was run against.',
+    }
+  );
+
+  const actionVariableContextQueryLabel = i18n.translate(
+    'xpack.stackAlerts.esQuery.actionVariableContextQueryLabel',
+    {
+      defaultMessage: 'The stringifed ES query.',
```

Comment:
Is there a better word for "stringified"?

Reply:
I changed it to "The string representation of the ES query." Does that sound better?